backend/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -269,98 +235,29 @@ public void addEmail(Email email) {
-
-                    aggregator.setStartTime(startTime);
-                    aggregator.setEndTimeKey(endTime);
-                    Action action = new Action();
-                    action.setContext(aggregator.getContext());
-                    action.setEvents(List.of());
-                    action.setAccountId(accountId);
-                    action.setApplication(application);
-                    action.setBundle(bundle);
-
-                    // We don't have a eventtype as this aggregates over multiple event types
-                    action.setEventType(null);
-                    action.setTimestamp(LocalDateTime.now(ZoneOffset.UTC));
-
-                    // We don't have any endpoint (yet) as this aggregates multiple endpoints
-                    Notification item = new Notification(action, null);
-
-                    return sendEmail(item, emailSubscriptionType).onItem().transformToMulti(notificationHistory -> Multi.createFrom().item(Tuple2.of(notificationHistory, aggregationKey)));
-                })
-                .onItem().transformToMultiAndConcatenate(result -> {
-                    if (delete) {
-                        return emailAggregationResources.purgeOldAggregation(aggregationKey, endTime)
-                                .toMulti()
-                                .onItem().transform(integer -> result);
-                    }
-
-                    return Multi.createFrom().item(result);
-                });
-                // Todo: If we want to save the NotificationHistory, this could be a good place to do so. We would probably require a special EndpointType
-                // .onItem().invoke(result -> { })
+    @Incoming("aggregation")
+    public void consumeEmailAggregations(String aggregationsJson) throws JsonProcessingException {
+        List<EmailAggregation> aggregations = objectMapper.readValue(aggregationsJson, new TypeReference<>() { });
+        Uni<List<EmailAggregation>> emailaggregations = Uni.createFrom().item(aggregations);
+        emailaggregations.onItem().transform(aggregationList -> {
+            aggregationList.forEach(emailAggregationItem -> {
+                Action action = new Action();
+                action.setContext(emailAggregationItem.getPayload().getMap());
+                action.setEvents(List.of());
+                action.setAccountId(emailAggregationItem.getAccountId());
+                action.setApplication(emailAggregationItem.getApplicationName());
+                action.setBundle(emailAggregationItem.getBundleName());
+
+                // We don't have a eventtype as this aggregates over multiple event types
+                action.setEventType(null);
+                action.setTimestamp(LocalDateTime.now(ZoneOffset.UTC));
+
+                // We don't have any endpoint (yet) as this aggregates multiple endpoints
+                Notification item = new Notification(action, null);
+
+                sendEmail(item, EmailSubscriptionType.DAILY);
```

Comment:
This returns a `Uni` with no subscription afterwards which mean the `Uni` may never be resolved (executed). Does it actually send an email?

Reply:
I don't know. I spend so much time in trying to test it but this still doesn't work.

Reply:
[This guide](https://smallrye.io/smallrye-mutiny/guides/testing) contains some information about testing `Uni`/`Multi` but it doesn't really explain how to use `UniAssertSubscriber` so a [quick look](https://github.com/search?l=Java&q=UniAssertSubscriber&type=Code) at the GitHub code from all repos that use `UniAssertSubscriber` can also be helpful. That's one way to test reactive code, but if we strictly consider the question of the emails being sent or not, I don't think `UniAssertSubscriber` is the easiest way to test that.

In notifications-backend, a webhook is called each time an email is sent so mocking the webhook server side with `mockserver` could be a solution to check that the emails are sent. We do that in [WebhookTest](https://github.com/RedHatInsights/notifications-backend/blob/master/backend/src/test/java/com/redhat/cloud/notifications/processors/webhook/WebhookTest.java) for example. But that's a bit overkill because I don't think the entire webhooks code should be tested here.

Mocking the `WebhookTypeProcessor webhookSender` attribute from `EmailSubscriptionTypeProcessor` with Mockito and verifying that the `doHttpRequest` method is called is probably enough to make sure emails are sent during the `EmailSubscriptionTypeProcessor#consumeEmailAggregations` execution.

Reply:
That's what I wanted to test. Doesn't work. Maybe someone else wants to give it a try?